static/js/settings.js

```diff
@@ -296,6 +326,89 @@ var SettingsScreen = {
     });
   },
 
+  showDiscoverAddonsScreen: function() {
+    this.backButton.href = '/settings/addons';
+    this.addonSettings.classList.remove('hidden');
+    this.addonMainSettings.classList.add('hidden');
+    this.addonDiscoverySettings.classList.remove('hidden');
+
+    const opts = {
+      headers: API.headers(),
+    };
+    fetch('/addons/discover', opts).then(function (response) {
+      return response.json();
+    }).then(function (body) {
+      if (!body) {
+        return;
+      }
+
+      const addonList = document.getElementById('discovered-addons-list');
+      addonList.innerHTML = '';
+
+      for (const addon of body) {
+        const li = document.createElement('li');
+        li.className = 'discovered-addon-item';
+
+        const headerDiv = document.createElement('div');
+        headerDiv.className = 'addon-settings-header';
+
+        const titleDiv = document.createElement('div');
+        titleDiv.innerText = addon.display_name;
+        titleDiv.className = 'addon-settings-name';
+
+        const descriptionDiv = document.createElement('div');
+        descriptionDiv.innerText = addon.description;
+        descriptionDiv.className = 'addon-settings-description';
+
+        const controlDiv = document.createElement('div');
+        controlDiv.className = 'addon-settings-controls';
```

Comment:
Nit: Can we come up with more semantic markup that just divs? At least span for the text labels.

Reply:
Yep, I'll switch them to spans.